### PR TITLE
chore: soft-delete underlying exported tasks.

### LINKED ISSUE #3303: Notebooks: When deleting a panel, soft delete the dependent task
When we delete a panel in Notebooks, we should soft delete the dependent task as the association would be broken with the Notebook / Panel. 

### Acceptance Criteria

* When a Notebook panel is deleted, any dependent tasks are soft deleted

### PR DESCRIPTION
**Pull request by subirjolly**
Closes #3303 

Set the status of underlying exported tasks from Notebooks to `inactive` asynchronously

### CODE REVIEW

**Review by drdelambre — CHANGES_REQUESTED** [review]
TLDR; right code, wrong place 

adding the patch request to deactivate the task is totally how to soft delete a task, but this code is panel type specific, which the shared sidebar component shouldn't know anything about. we need some abstraction here to decouple the shared interaction and what happens specifically for the schedule panel. what we're trying to avoid here is a giant switch statement in all of the shared actions becoming a pattern and having all panels have to register a universal action (to reduce the problem of having a panel forget to add one)

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 56 | :red_circle:
accessibility | 97 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=3696)

File: src/flows/context/flow.current.tsx [hunk 1]
```diff
@@ -304,6 +305,19 @@ export const FlowProvider: FC = ({children}) => {
   }
 
   const removePipe = (id: string) => {
+    if (currentFlow.data.byID[id].type === 'schedule') {
```
Comment (drdelambre) [inline src/flows/context/flow.current.tsx:308]: Looks like you just moved the same problem somewhere else

File: src/flows/context/flow.current.tsx [hunk 2]
```diff
@@ -304,6 +305,14 @@ export const FlowProvider: FC = ({children}) => {
   }
 
   const removePipe = (id: string) => {
+    const definition = PIPE_DEFINITIONS[currentFlow.data.byID[id].type]
+
+    // FIXME: Should i do this check or should i make sure that
```
Comment (drdelambre) [inline src/flows/context/flow.current.tsx:310]: Remove this comment

File: src/flows/pipes/Schedule/remove.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import {patchTask} from 'src/client'
```
Comment (drdelambre) [inline src/flows/pipes/Schedule/remove.ts:1]: Nit: this should be in the definition, no need for another file

**Review by drdelambre — APPROVED** [review]
Clean up the comment

File: src/flows/pipes/Schedule/remove.ts (hunk 3 above)
Comment (subirjolly) [reply src/flows/pipes/Schedule/remove.ts]: updated

**Pull request merged**